docs/extensibility/custom-non-container-resource.md

```diff
@@ -0,0 +1,145 @@
+
+## Key components
+
+Non-container custom resources use these key Aspire services:
+
+- **<xref:Aspire.Hosting.Lifecycle.IDistributedApplicationLifecycleHook>**: Hook into app startup/shutdown. For more information, see [App Host life cycle events](../app-host/eventing.md#app-host-life-cycle-events).
+- **<xref:Microsoft.Extensions.Logging.ILogger>**: Standard .NET logging that appears in console and dashboard
+
+> [!NOTE]
+> Advanced dashboard integration is possible using services like `ResourceNotificationService` and `ResourceLoggerService` for real-time status updates and log streaming. These APIs provide richer dashboard experiences but require more complex implementation.
+
+## Example: HTTP proxy resource
+
+This example creates an in-process HTTP proxy resource that demonstrates the core concepts of lifecycle management and logging integration with the Aspire dashboard.
+
+### Define the resource
+
+First, create the resource class:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxy.Hosting/HttpProxyResource.cs":::
+
+The resource implements <xref:Aspire.Hosting.ApplicationModel.IResource> and includes properties for the proxy configuration.
+
+### Create the extension method
+
+Next, create the builder extension:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxy.Hosting/HttpProxyResourceBuilderExtensions.cs":::
+
+This extension method adds the resource to the application model and configures an HTTP endpoint.
+
+### Implement lifecycle management
+
+Create a lifecycle hook to manage the proxy:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxy.Hosting/HttpProxyLifecycleHook.cs":::
+
+The lifecycle hook:
+
+1. **Manages lifecycle**: Starts services when resources are created
+2. **Integrates logging**: Uses standard .NET logging that appears in the Aspire dashboard
+3. **Handles background tasks**: Runs long-running services in background tasks
+4. **Provides resource management**: Manages resources like HTTP listeners and cleanup
+
+### Register the lifecycle hook
+
+The extension method automatically registers the lifecycle hook:
+
+```csharp
+public static IResourceBuilder<HttpProxyResource> AddHttpProxy(
+    this IDistributedApplicationBuilder builder,
+    string name,
+    string targetUrl,
+    int? port = null)
+{
+    var resource = new HttpProxyResource(name, targetUrl);
+    
+    // Register the lifecycle hook for this resource type
+    builder.Services.TryAddSingleton<HttpProxyLifecycleHook>();
+    builder.Services.AddLifecycleHook<HttpProxyLifecycleHook>();
+
+    return builder.AddResource(resource)
+                  .WithHttpEndpoint(port: port, name: "http");
+}
+```
+
+### Use the resource
+
+Now you can use the proxy in your app host:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxySample.AppHost/Program.cs":::
+
+## Dashboard integration
+
+The resource integrates with the Aspire dashboard through:
```

Comment:
I'm not a fan of this, it's an opening statement that's terminated with the `:`. I was expecting a diagram, bullet list, or something... it took me a bit to realize that it was intended to be the subsections. I'd prefer strong details about the dashboard integration.